src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -38,7 +37,7 @@ internal sealed class RSession : IRSession, IRCallbacks {
         public event EventHandler<EventArgs> Disconnected;
         public event EventHandler<EventArgs> Disposed;
         public event EventHandler<EventArgs> DirectoryChanged;
-        public event EventHandler<EventArgs> BeforePackagesInstalled;
+        public event AsyncEventHandler<EventArgs> BeforePackagesInstalledAsync;
```

Comment:
This is complicated. `IPackageIndex` is in Support that depends on Components but callback is in `Components` and knows nothing about `IPackageIndex`. I can try and move interfaces but change will be huge